### PR TITLE
Tablet screenshot tests

### PR DESCRIPTION
**Pull request by a-szotyori**
Moved the screenshot tests to the Screen API example app module and extended it to take screenshots of all the screens in portrait and landscape orientations.

### How to test
Run the `ScreenshotsTest` on a device or emulator. The screenshots will be saved to the `<SDCard>/test-screenshots` folder on the device/emulator.

### Merging
Automatic.

### Ticket 
Issue #54

### CODE REVIEW

File: ginivision/src/androidTest/java/net/gini/android/vision/uiautomator/ScreenshotsTest.java [hunk 1]
```diff
@@ -1,86 +0,0 @@
...
-
-    private static final String SCREEN_API_EXAMPLE_APP = "net.gini.android.vision.screenapiexample";
-
-    private UiDevice mDevice;
-
-    @Before
-    public void setUp() throws Exception {
-        mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
-        assertThat(mDevice).isNotNull();
-        // Start from the home screen
```
Comment (kikettas) [inline ginivision/src/androidTest/java/net/gini/android/vision/uiautomator/ScreenshotsTest.java]: Just a doubt. Why is it needed to start from the home screen?

File: screenapiexample/src/androidTest/java/net/gini/android/vision/screen/testhelper/PermissionsHelper.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+     * @return command's standard out
+     */
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    @NonNull
+    private static String executeShellCommand(final String command) {
+        UiAutomation uiAutomation = getInstrumentation().getUiAutomation();
+        try (ParcelFileDescriptor parcelFd = uiAutomation.executeShellCommand(command);
+             FileReader fileReader = new FileReader(parcelFd.getFileDescriptor());
+             BufferedReader reader = new BufferedReader(fileReader)) {
+            StringBuilder stringBuilder = new StringBuilder();
```
Comment (kikettas) [inline screenapiexample/src/androidTest/java/net/gini/android/vision/screen/testhelper/PermissionsHelper.java]: Has this the correct indentation? It is not easy to read.

File: screenapiexample/src/androidTest/java/net/gini/android/vision/screen/screenshots/ScreenshotsTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    private static final String SCREEN_API_EXAMPLE_APP = "net.gini.android.vision.screenapiexample";
+
+    private UiDevice mDevice;
+
+    @Before
+    public void setUp() throws Exception {
+        mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        assertThat(mDevice).isNotNull();
+        // Start from the home screen
```
Comment (kikettas) [inline screenapiexample/src/androidTest/java/net/gini/android/vision/screen/screenshots/ScreenshotsTest.java]: Just a doubt. Why is it needed to start from the home screen?

Comment (a-szotyori) [reply screenapiexample/src/androidTest/java/net/gini/android/vision/screen/screenshots/ScreenshotsTest.java]: The testing framework is not starting the application automatically, but the gradle tasks executing the tests installs the app before test execution. In order to start each test from the same state we go to the home screen. Each test method then launches the app when required.

File: screenapiexample/src/androidTest/java/net/gini/android/vision/screen/testhelper/PermissionsHelper.java (hunk 2 above)
Comment (a-szotyori) [reply screenapiexample/src/androidTest/java/net/gini/android/vision/screen/testhelper/PermissionsHelper.java]: Strangely it has correct indentation... due to the [try with resources statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) needing multiple lines the indentation looks off. Actually this is a case where curly braces on a new line would make it clearer 🤔...

**Comment by a-szotyori** [comment]
Thank you for your review @kikettas, I like the [Page Object pattern](https://martinfowler.com/bliki/PageObject.html) as it abstracts interface interaction and assertion away in a logical way and makes the tests very readable.

File: screenapiexample/src/androidTest/java/net/gini/android/vision/screen/testhelper/PermissionsHelper.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import android.os.Build;
+import android.os.ParcelFileDescriptor;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class PermissionsHelper {
```
Comment (EarlOfEgo) [inline screenapiexample/src/androidTest/java/net/gini/android/vision/screen/testhelper/PermissionsHelper.java]: FYI: The Android Testing Support Library 1.0 comes with a [GrantPermissionRule](https://developer.android.com/reference/android/support/test/rule/GrantPermissionRule.html)

**Pull request merged**